### ISSUE TITLE
Could not find matching constructor for: org.grails.datastore.gorm.bootstrap.AbstractDatastoreInitializer(groovy.util.ConfigObject, java.util.ArrayList)

### ISSUE THREAD

**Issue by PhilipWu** [grails/grails-data-mapping#711]
hi

I know this issue has been reported before in #610, but the issue is still happening for me. My envinroment is as follows:

Grails: 2.5.4
MongoDB plugin: 5.0.5.RELEASE

Is nobody else seeing this? How are other people using this plugin? Using grails 3.x?

I even tried this on a fresh app by using the following

`grails create-app test-app`

Then I modify the BuildConfig.groovy and the plugin as follows:

`runtime ":mongodb:5.0.5.RELEASE"`

Then try running the app:

`grails run-app`

You will see the following error:

`2016-05-04 09:54:47,087 [localhost-startStop-1] ERROR context.GrailsContextLoaderListener  - Error initializing the application: Could not find matching constructor for: org.grails.datastore.gorm.bootstrap.AbstractDatastoreInitializer(groov
Message: Could not find matching constructor for: org.grails.datastore.gorm.bootstrap.AbstractDatastoreInitializer(groovy.util.ConfigObject, java.util.ArrayList)
    Line | Method
->>   33 | doCall                    in MongodbGrailsPlugin$_closure1
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
|    754 | invokeBeanDefiningClosure in grails.spring.BeanBuilder
|    584 | beans . . . . . . . . . . in     ''
|    527 | invokeMethod              in     ''
|    262 | run . . . . . . . . . . . in java.util.concurrent.FutureTask
|   1145 | runWorker                 in java.util.concurrent.ThreadPoolExecutor
|    615 | run . . . . . . . . . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run                       in java.lang.Thread`

**Comment by graemerocher** [comment]
Tried the same steps and works fine for me

**Comment by moskiteau** [comment]
I can reproduce this bug with 2.5.4 and 5.0.x

**Comment by graemerocher** [comment]
@moskiteau upload an example that reproduces it then please

**Comment by moskiteau** [comment]
[bug_issue_711.zip](https://github.com/grails/grails-data-mapping/files/330266/bug_issue_711.zip)

**Comment by graemerocher** [comment]
This is a dependency plugin `mongodb` 5.x requires hibernate4 5.x. After defining these dependencies:

```
        compile ":hibernate4:5.0.8.RELEASE"
        compile ":mongodb:5.0.8.RELEASE"
```

It works fine

**Comment by moskiteau** [comment]
Great, thanks!

What is the latest mongo plugins compatible with hibernate 3.x ?

There is a lack of info on thoses compatibilities, a dependency matrix would be great

**Comment by graemerocher** [comment]
there is no hibernate3 compatible version and there won't be for Grails 2.x

Hibernate 3 support will also be dropped entirely in GORM 6 released with Grails 3.2
